### PR TITLE
Add setup.py and basic project structure

### PR DESCRIPTION
**Pull request by tomas-mm**
Just some initial project structure and settings.

Maybe the factory classes with just static methods could be converted into functions (in python they are 'first-class citizens')

### CODE REVIEW

File: tests/resources/stages/Tokenizer/metadata/part-00000 [hunk 1]
```diff
@@ -1 +1 @@
-{"class":"org.apache.spark.ml.feature.Tokenizer","timestamp":1588797806167,"sparkVersion":"2.4.0","uid":"Tokenizer_7e123b887c49","paramMap":{"outputCol":"words","inputCol":"text"},"defaultParamMap":{"outputCol":"Tokenizer_7e123b887c49__output"}}
+{"class":"org.apache.spark.ml.feature.Tokenizer","timestamp":1589129933974,"sparkVersion":"2.4.5","uid":"Tokenizer_1bfbddaa18be","paramMap":{"outputCol":"words","inputCol":"text"},"defaultParamMap":{"outputCol":"Tokenizer_1bfbddaa18be__output"}}
```
Comment (tomas-mm) [inline tests/resources/stages/Tokenizer/metadata/part-00000:1]: probably this file should not be pushed

Comment (lluccardoner) [reply tests/resources/stages/Tokenizer/metadata/part-00000:1]: I have a TODO to delete files created in the tests

**Review by lluccardoner — APPROVED** [review]

File: src/ml_pipeline/examples/pipeline.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from pyspark.sql import SparkSession
 
-from src.step_factory import StepFactory
+from ml_pipeline
```
Comment (lluccardoner) [inline src/ml_pipeline/examples/pipeline.py]: Repeated?

File: src/ml_pipeline/examples/pipeline_from_config.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from pyspark.sql import SparkSession
 
 from config_factory import ConfigFactory
-from src.step_factory import StepFactory
+from src import StepFactory
```
Comment (lluccardoner) [inline src/ml_pipeline/examples/pipeline_from_config.py]: Should this be `from ml_pipeline import StepFactory`?

File: src/ml_pipeline/examples/pipeline.py (hunk 2 above)
Comment (tomas-mm) [reply src/ml_pipeline/examples/pipeline.py]: yep

File: src/ml_pipeline/examples/pipeline_from_config.py (hunk 3 above)
Comment (tomas-mm) [reply src/ml_pipeline/examples/pipeline_from_config.py]: exactly

File: src/ml_pipeline/examples/pipeline.py (hunk 2 above)
Comment (tomas-mm) [reply src/ml_pipeline/examples/pipeline.py]: fixed!

File: src/ml_pipeline/examples/pipeline_from_config.py (hunk 3 above)
Comment (tomas-mm) [reply src/ml_pipeline/examples/pipeline_from_config.py]: done!

**Pull request merged**